src/main/java/formflow/library/interceptors/DataRequiredInterceptor.java

```diff
@@ -0,0 +1,117 @@
+package formflow.library.interceptors;
+
+import formflow.library.config.FlowConfiguration;
+import formflow.library.exceptions.LandmarkNotSetException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpSession;
+import java.util.List;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.core.Ordered;
+import org.springframework.stereotype.Component;
+import org.springframework.util.AntPathMatcher;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+/**
+ * This interceptor prevents users from jumping to random pages in a flow.
+ */
+@Component
+@Slf4j
+@ConditionalOnProperty(name = "form-flow.session-continuity-interceptor.enabled", havingValue = "true")
+public class DataRequiredInterceptor implements HandlerInterceptor, Ordered {
```

Comment:
For naming: I can see that because we require both session and submission id, that it's data that's required, but maybe this would be more aptly named `SessionContinuityInterceptor` to make it immediately recognizable (to programmers) that it's related to the configuration parameter?

Reply:
I like it!  I'll get that changed